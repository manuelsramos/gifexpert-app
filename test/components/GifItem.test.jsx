import { render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe('Prueba en GifItem', () => {
    const title = 'title';
    const url = 'https://dosods.com'


    test('should match with snapshot', () => {


        const { container } = render(<GifItem title={title} url={url} />);

        expect(container).toMatchSnapshot();
    })

    test('should show the image with URL and ALT', () => {

        render(<GifItem title={title} url={url} />);
        /*         expect(screen.getByRole('img').src).toBe(url);
                expect(screen.getByRole('img').alt).toBe(title); */
        /*   console.log(screen.getByRole('img')); */

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });


    test('should show component title', () => {
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();


    })
});