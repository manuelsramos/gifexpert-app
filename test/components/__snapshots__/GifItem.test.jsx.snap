// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en GifItem should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="https://dosods.com"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
